### PR TITLE
Mecha medical and other_equipment refactor

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

fucking mechcode

## Why It's Good For The Game

who needs soul anyway

## Changelog
:cl:
refactor: Mecha medical and other_equipment refactor
/:cl:

### CODE REVIEW

File: code/modules/projectiles/ammunition/special/syringe.dm [hunk 1]
```diff
@@ -20,7 +20,16 @@
...
+	else if(istype(loc, /obj/item/mecha_parts/mecha_equipment/medical/syringe_gun))
+		var/obj/item/mecha_parts/mecha_equipment/medical/syringe_gun/SG = loc
+		var/obj/item/reagent_containers/syringe/S = SG.syringes[1]
+		SG.reagents.trans_to(BB, min(S.volume, SG.reagents.total_volume), transfered_by = user)
+		BB.name = S.name
+		var/obj/projectile/bullet/dart/D = BB
+		D.piercing = S.proj_piercing
+		LAZYREMOVE(SG.syringes, S)
+		qdel(S)
+	return ..()
```
Comment (Timberpoes) [inline code/modules/projectiles/ammunition/special/syringe.dm:32]: image

File: code/modules/vehicles/mecha/equipment/tools/other_tools.dm [hunk 2]
```diff
@@ -327,25 +325,19 @@
...
-			if(pow_chan)
-				var/delta = min(10 * delta_time, chassis.cell.maxcharge-cur_charge)
-				chassis.give_power(delta)
-				A.use_power(delta*coeff, pow_chan)
+	var/area/A = get_area(chassis)
+	var/pow_chan = get_chassis_area_power(A)
+	if(pow_chan)
+		var/delta = min(10 * delta_time, chassis.cell.maxcharge-cur_charge)
+		chassis.give_power(delta)
+		A.use_power(delta*coeff, pow_chan)
```
Comment (Timberpoes) [inline code/modules/vehicles/mecha/equipment/tools/other_tools.dm:340]: Ditto

**Review by Timberpoes — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob_movement.dm [hunk 3]
```diff
@@ -268,7 +268,8 @@
   * You can move in space if you have a spacewalk ability
   */
 /mob/Process_Spacemove(movement_dir = 0)
-	if(HAS_TRAIT(src, TRAIT_SPACEWALK) || ..())
+	. = ..()
+	if(HAS_TRAIT(src, TRAIT_SPACEWALK) || .)
```
Comment (Rohesie) [inline code/modules/mob/mob_movement.dm:272]:
```suggestion
	if(. || HAS_TRAIT(src, TRAIT_SPACEWALK))
```

File: code/modules/projectiles/ammunition/special/syringe.dm [hunk 4]
```diff
@@ -20,7 +20,16 @@
...
-	..()
+	else if(istype(loc, /obj/item/mecha_parts/mecha_equipment/medical/syringe_gun))
+		var/obj/item/mecha_parts/mecha_equipment/medical/syringe_gun/SG = loc
+		var/obj/item/reagent_containers/syringe/S = SG.syringes[1]
+		SG.reagents.trans_to(BB, min(S.volume, SG.reagents.total_volume), transfered_by = user)
+		BB.name = S.name
+		var/obj/projectile/bullet/dart/D = BB
+		D.piercing = S.proj_piercing
+		LAZYREMOVE(SG.syringes, S)
+		qdel(S)
```
Comment (Rohesie) [inline code/modules/projectiles/ammunition/special/syringe.dm:31]:
```suggestion
		var/obj/item/mecha_parts/mecha_equipment/medical/syringe_gun/syringe_gun = loc
		var/obj/item/reagent_containers/syringe/loaded_syringe = syringe_gun.syringes[1]
		syringe_gun.reagents.trans_to(BB, min(loaded_syringe.volume, syringe_gun.reagents.total_volume), transfered_by = user)
		BB.name = loaded_syringe.name
		var/obj/projectile/bullet/dart/shot_dart = BB
		shot_dart.piercing = loaded_syringe.proj_piercing
		LAZYREMOVE(syringe_gun.syringes, loaded_syringe)
		qdel(loaded_syringe)
```

File: code/modules/vehicles/mecha/equipment/tools/other_tools.dm [hunk 5]
```diff
@@ -27,105 +28,104 @@
...
-	var/turf/target_turf = pick(L)
+	var/list/validturfs = list()
+	var/turf/ourturf = get_turf(src)
+	for(var/turf/T in get_area_turfs(targetarea.type))
+		if(T.density || chassis.z != T.z)
+			continue
+		for(var/obj/O in T)
+			if(!O.density)
+				continue
+			validturfs += T
```
Comment (Rohesie) [inline code/modules/vehicles/mecha/equipment/tools/other_tools.dm:53]:
```suggestion
	for(var/t in get_area_turfs(targetarea.type))
		var/turf/evaluated_turf = t
		if(evaluated_turf.density || chassis.z != evaluated_turf.z)
			continue
		for(var/obj/evaluated_obj in evaluated_turf)
			if(!evaluated_obj.density)
				continue
			validturfs += evaluated_turf
```

**Pull request merged**